tests/test_parametric_shapes/test_ExtrudeMixedShape.py

```diff
@@ -117,108 +85,61 @@ def test_cut_volume(self):
+        can be exported with the correct suffix using the export_stl method."""
 
-        test_shape = ExtrudeMixedShape(
-            points=[
-                (10, 20, "straight"),
-                (10, 10, "straight"),
-                (20, 10, "circle"),
-                (22, 15, "circle"),
-                (20, 20, "straight"),
-            ],
-            distance=10,
-        )
-        os.system("rm tests/test.stl")
-        test_shape.export_stl("tests/test.stl")
-        assert Path("tests/test.stl").exists() is True
-        os.system("rm tests/test.stl")
-        test_shape.export_stl("tests/test")
-        assert Path("tests/test.stl").exists() is True
-        os.system("rm tests/test.stl")
+        os.system("rm filename.stl")
+        self.test_shape.export_stl("filename.stl")
+        assert Path("filename.stl").exists() is True
+        os.system("rm filename.stl")
+        self.test_shape.export_stl("filename")
+        assert Path("filename.stl").exists() is True
+        os.system("rm filename.stl")
 
     def test_rotation_angle(self):
-        """creates an extruded shape with a rotation_angle < 360 and checks that the
-        correct cut is performed and the volume is correct"""
-
-        test_shape = ExtrudeMixedShape(
-            points=[
-                (10, 20, "straight"),
-                (10, 10, "straight"),
-                (20, 10, "circle"),
-                (22, 15, "circle"),
-                (20, 20, "straight"),
-            ],
-            distance=10,
-            azimuth_placement_angle=[45, 135, 225, 315]
-        )
-        test_volume = test_shape.volume
-
-        test_shape.rotation_angle = 180
-
-        assert test_shape.volume == pytest.approx(test_volume * 0.5)
-
-    def test_extrude_both(self):
-        """test that the volume with extrude_both=True is twice as when
-        extrude_both=False
-        """
-        test_shape = ExtrudeMixedShape(
-            points=[
-                (10, 20, "straight"),
-                (10, 10, "spline"),
-                (20, 10, "circle"),
-                (22, 15, "circle"),
-                (20, 20, "straight"),
-            ],
-            distance=10,
-        )
-        test_volume_double_extrude = test_shape.volume
-        test_shape.extrude_both = False
-
-        assert test_shape.volume == \
-            pytest.approx(test_volume_double_extrude * 0.5)
+        """Creates an ExtrudeMixedShape with a rotation_angle < 360 and checks that
+        the correct cut is performed and the volume is correct."""
+
+        self.test_shape.azimuth_placement_angle = [45, 135, 225, 315]
+        test_volume = self.test_shape.volume
+        self.test_shape.rotation_angle = 180
+        assert self.test_shape.volume == pytest.approx(
+            test_volume * 0.5, rel=0.01)
+
+    # def test_extrude_both(self):
+    #     """test that the volume with extrude_both=True is twice as when
```

Comment:
perhaps this can be included but it checks that the volumes are similar size

Reply:
Yes, this test should be included. At the moment, extrude_both=True generates a solid with double the volume of extrude_both=False meaning somewhere we are still dividing extrusion distance by 2 when we shouldn't

Reply:
ah thanks for that info. 

If the test is included will it help find the Shapes that are misbehaving. Perhaps this should be included for all extrude shapes

Reply:
This will be fixed on a new branch after this PR has been merged